addons/google_recaptcha/static/src/js/recaptcha.js

```diff
@@ -0,0 +1,88 @@
+odoo.define('google_recaptcha.ReCaptchaV3', function (require) {
+"use strict";
+
+const ajax = require('web.ajax');
+const Class = require('web.Class');
+const core = require('web.core');
+
+const qweb = core.qweb;
+const _t = core._t;
+
+const ReCaptcha = Class.extend({
+    /**
+     * @override
+     */
+    init: function () {
+        this._publicKey = odoo.reCaptchaPublicKey;
+    },
+    /**
+     * Loads the recaptcha libraries.
+     *
+     * @returns {Promise|boolean} promise if libs are loading else false if the reCaptcha key is empty.
+     */
+    loadLibs: function () {
+        if (this._publicKey) {
+            this._recaptchaReady = ajax.loadJS(`https://www.recaptcha.net/recaptcha/api.js?render=${this._publicKey}`)
+                .then(() => new Promise(resolve => window.grecaptcha.ready(() => resolve())));
+            const legalTermsReady = ajax.loadXML('/google_recaptcha/static/src/xml/recaptcha.xml', qweb);
+            this._libsLoaded = Promise.all([this._recaptchaReady, legalTermsReady]).then(() => !!document.querySelector('.grecaptcha-badge'));
+            return this._libsLoaded;
+        }
+        return false;
+    },
+    /**
+     * Adds the required recaptcha legal therms or an error if invalid public key after this elem
+     *
+     * @param {HTMLElement} options.legalTargetEl
+     * @returns {Promise|boolean}
+     */
+    insertLegalTermsAfter: async function (legalTargetEl) {
+        if (!this._publicKey) {
+            return false;
+        }
+        const libsLoaded = await this._libsLoaded;
+        const template = document.createElement('template');
+        template.innerHTML = qweb.render(`recaptcha.${libsLoaded ? 'legal_terms' : 'error'}`);
```

Comment:
Useless construction of the template name. If I want to grep recaptcha.error, I won't find it and I will have to grep "error".

Reply:
keep it simple ;)
(libsLoaded ? `recaptcha.legal_terms' : 'recaptcha.legal_terms')